src/main/java/seedu/linkedout/model/ModelManager.java

```diff
@@ -144,7 +150,52 @@ public boolean equals(Object obj) {
         ModelManager other = (ModelManager) obj;
         return linkedout.equals(other.linkedout)
                 && userPrefs.equals(other.userPrefs)
-                && filteredApplicants.equals(other.filteredApplicants);
+                && filteredApplicants.equals(other.filteredApplicants)
+                && sortedApplicants.equals(other.sortedApplicants);
+    }
+
+    //=========== Sorted Applicant List Accessors =============================================================
+
+    @Override
+    public ObservableList<Applicant> getSortedApplicantList() {
+        return sortedApplicants;
+    }
+
+    //=========== Search Applicant List Accessors =============================================================
+
+    @Override
+    public void updateSearchedApplicantList(List<KeywordsPredicate> predicates) {
+        requireNonNull(predicates);
+        Predicate<Applicant> keywordPredicate = combinePredicates(predicates);
+        updateFilteredApplicantList(keywordPredicate);
+        sortedApplicants.setComparator((a1, a2) -> numberOfKeywordMatches(a2, predicates)
+                - numberOfKeywordMatches(a1, predicates));
     }
 
+    /**
+     * Returns the number of matched input keywords with an applicant
+     * @param applicant
+     * @param predicates list of predicates
+     * @return number of keywords matched
+     */
+    private static int numberOfKeywordMatches (Applicant applicant, List<KeywordsPredicate> predicates) {
+        int matchedNumber = 0;
```

Comment:
Need to check if null or if list is empty?

Reply:
if list is empty it will not enter the for loop, and list will not be null since the list is created from getArrrayOfKeywords and splitKeywordsWithWhitespace(in SearchCommandParser) which already check if keywords is empty or not

Reply:
Okays nice